packages/react-api/src/api/model.js

```diff
@@ -66,15 +66,46 @@ export function executeModel(props) {
     filtersLogicalOperator
   };
 
-  // API supports multiple filters, we apply it only to geoColumn
-  const spatialFilters = spatialFilter
-    ? {
-        [source.geoColumn ? source.geoColumn : DEFAULT_GEO_COLUMN]: spatialFilter
+  let spatialFilters;
+  if (spatialFilter) {
+    let spatialDataType = source.spatialDataType;
+    let spatialDataColumn = source.spatialDataColumn;
+
+    if (!spatialDataType || !spatialDataColumn) {
+      if (source.geoColumn) {
+        const parsedGeoColumn = source.geoColumn ? source.geoColumn.split(':') : [];
```

Comment:
Checking `source.geoColumn` redundantly here – probably one or the other isn't required?